lib/manager.js

```diff
@@ -401,6 +401,7 @@ Manager.prototype.destroy = function(socket){
 Manager.prototype.packet = function(packet){
   debug('writing packet %j', packet);
   var self = this;
+  if (packet.query && packet.type == 0) packet.nsp += '?' + packet.query;
```

Comment:
That's possible, but that would require changes to Socket.io but the corresponding [PR](https://github.com/socketio/socket.io/pull/2422) has already been merged.